lib/identity_config.rb

```diff
@@ -4,7 +4,20 @@ class << self
   end
 
   CONVERTERS = {
-    string: proc { |value| value.to_s },
+    # Allows loading a string configuration from a system environment variable
+    # ex: To read DATABASE_HOST from system environment for the database_host key
+    # database_host: ['env', 'DATABASE_HOST']
+    # To use a string value directly, you can specify a string explicitly:
+    # database_host: 'localhost'
+    string: proc do |value|
+      if value.is_a?(Array) && value.length == 2 && value.first == 'env'
+        ENV.fetch(value[1])
+      elsif value.is_a?(String)
+        value
+      else
+        raise 'invalid system environment configuration value'
+      end
```

Comment:
I feel like I suggested this last time, but I think we can use array destructuring in the params to give clearer names?
```suggestion
    string: proc do |(key_or_value, env_var)|
      if first == 'env' && env_var.present?
        ENV.fetch(env_var)
      elsif key_or_value.is_a?(String)
        key_or_value
      else
        raise 'invalid system environment configuration value'
      end
```

Reply:
i am in favor of the clearer names!